### PR TITLE
HARP-10793: Add styleSetName and category to MapAnchors

### PR DESCRIPTION
**Pull request by ninok**

### CODE REVIEW

File: @here/harp-mapview/lib/MapAnchors.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+     * Add a [[MapAnchor]].
+     * @param mapAnchor [[MapAnchor]] instance to add.
+     */
+    add(mapAnchor: MapAnchor) {
+        this.m_anchors.push(mapAnchor);
+    }
+
+    /**
+     * Remove a [[MapAnchor]].
+     * @param mapAnchor [[MapAnchor]] instance to remove.
```
Comment (robertoraggi) [inline @here/harp-mapview/lib/MapAnchors.ts:87]: you are missing the - after `@param`

File: @here/harp-mapview/lib/MapAnchors.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+     */
+    category?: string;
+};
+
+export class MapAnchors {
+    private m_anchors: MapAnchor[] = [];
+
+    /**
+     *
+     */
```
Comment (robertoraggi) [inline @here/harp-mapview/lib/MapAnchors.ts:72]: missing documentation.

File: @here/harp-mapview/lib/MapAnchors.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+     * The category of this style.
+     *
+     * @remarks
+     * This property is used together with [[Theme.priorities]] to compute the render
+     * order of this mao object.
+     */
+    category?: string;
+};
+
+export class MapAnchors {
```
Comment (robertoraggi) [inline @here/harp-mapview/lib/MapAnchors.ts:67]: missing docs

File: @here/harp-mapview/lib/MapAnchors.ts [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    styleSet?: string;
+
+    /**
+     * The category of this style.
+     *
+     * @remarks
+     * This property is used together with [[Theme.priorities]] to compute the render
+     * order of this mao object.
+     */
+    category?: string;
```
Comment (robertoraggi) [inline @here/harp-mapview/lib/MapAnchors.ts:64]: we could use a reserved styleSet/category combination to identify the objects that need to be overlayed on top of the map scene (after text, markers and extruded buildings are rendered), alternatively we may need to add a property to MapAnchor that can be used to specify the scene/pass.

Comment (ninok) [reply @here/harp-mapview/lib/MapAnchors.ts:64]: yes, I wanted to discuss that as well. I will add the overlay scene in a separate PR though.

**Pull request merged**